### PR TITLE
🔨 Enhance diagnostics info in e2e testing

### PR DESCRIPTION
**Pull request by pcrespov**
<!-- Title Annotations:

  WIP: work in progress
  🐛    Fix a bug.
  ✨    Introduce new features.
  🎨    Enhance existing feature.
  ♻️    Refactor code.
  🚑️    Critical hotfix.
  ⚗️    Perform experiments.
  ⬆️    Upgrade dependencies.
  📝    Add or update documentation.
  🔨    Add or update development scripts.
  🔒️    Fix security issues.
  ⚠️    Changes in ops configuration etc. are required before deploying.
        [ Please add a link to the associated ops-issue or PR, such as in https://github.com/ITISFoundation/osparc-ops-environments or https://git.speag.com/oSparc/osparc-infra ]
  🗃️    Database table changed (relevant for devops).
  👽️    Public API changes (meaning: dev features are moved to being exposed in production)
  🚨    Do manual testing when deployed

or from https://gitmoji.dev/
-->

## What do these changes do?

This PR does some changes in the e2e-playwritght tests
- adds test report on failures that **includes graylog URL** within the time interval of the test. This will aid to diagnostic the state of the backend
- renames e2e-playwright tests to follow naming convention

## Related issue/s

- e2e maintenance


## How to test

This is how it would display in case of failure
```
platform linux -- Python 3.10.13, pytest-8.2.2, pluggy-1.5.0
rootdir: /home/crespo/repos/sandbox-python/test_pytest
plugins: asyncio-0.23.7
asyncio: mode=strict
collected 1 item                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              

test_it.py 
Diagnostics for test_it:
   test_name : test_it
   test_location : ('test_it.py', 0, 'test_it')
   api_host : https://api.osparc.io
   graylog_url : https://monitoring.osparc.io/graylog/search?from=2024-06-19T09:58:42.978364Z&to=2024-06-19T09:58:42.989394Z
F                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            [100%]

================================================================================================================================================================================================================================================== FAILURES ===================================================================================================================================================================================================================================================
___________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________ test_it ___________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________

    def test_it():
    
        print("foo")
>       assert False
E       assert False

test_it.py:4: AssertionError
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
foo
=========================================================================================================================================================================================================================================== short test summary info ===========================================================================================================================================================================================================================================
FAILED test_it.py::test_it - assert False
```


## Dev-ops checklist
None

### CODE REVIEW

File: tests/e2e-playwright/tests/conftest.py [hunk 1]
```diff
@@ -94,6 +95,70 @@ def pytest_addoption(parser: pytest.Parser) -> None:
     )
 
 
+# Dictionary to store start times of tests
+_test_start_times = {}
+
+
+def _utc_now() -> datetime.datetime:
+    return datetime.datetime.now(tz=datetime.timezone.utc)
```
Comment (GitHK) [inline tests/e2e-playwright/tests/conftest.py:103]: minor: maybe use arrow, like we do in the rest of the codebase?

**Review by GitHK — APPROVED** [review]

File: tests/e2e-playwright/tests/conftest.py [hunk 2]
```diff
@@ -94,6 +95,70 @@ def pytest_addoption(parser: pytest.Parser) -> None:
...
+    _test_start_times[item.name] = _utc_now()
+
+
+def _construct_graylog_url(
+    api_host: str, start_time: datetime.datetime, end_time: datetime.datetime
+) -> str:
+    base_url = api_host.replace("api.", "monitoring.", 1).rstrip("/")
+    url = f"{base_url}/graylog/search"
+    start_time_str = start_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+    end_time_str = end_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
```
Comment (sanderegg) [inline tests/e2e-playwright/tests/conftest.py:119]: isoformat does not work?

**Review by sanderegg — APPROVED** [review]
very nice and simple addition.

Question: will it print it on failure only? and when? at the end or at the beginning of the logs? or both?

**Review by odeimaiz — APPROVED** [review]
merci

File: tests/e2e-playwright/tests/conftest.py (hunk 2 above)
Comment (pcrespov) [reply tests/e2e-playwright/tests/conftest.py:119]:
```cmd
>>> jetzt.isoformat()
'2024-06-20T08:48:13.605788+00:00'
>>> jetzt.isoformat() == jetzt.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
False
>>> jetzt.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
'2024-06-20T08:48:13.605788Z'
```

Comment (pcrespov) [reply tests/e2e-playwright/tests/conftest.py:119]: these are used for the urls

File: tests/e2e-playwright/tests/conftest.py (hunk 1 above)
Comment (pcrespov) [reply tests/e2e-playwright/tests/conftest.py:103]: ok

**Comment by pcrespov** [comment]
> Question: will it print it on failure only? and when? at the end or at the beginning of the logs? or both?
@sanderegg in the description I show an example of how it prints. It does it on failure only and just after the definition and before the failure reported.

**Pull request merged**